src/main/java/core/basesyntax/service/amount/SubtractAmount.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.service.amount;
+
+import core.basesyntax.dao.FruitsDao;
+import core.basesyntax.dao.FruitsDaoImpl;
+import core.basesyntax.model.FruitRecord;
+
+public class SubtractAmount implements AmountHandler {
+    @Override
+    public void modifyAmount(FruitRecord newFruit) {
+        FruitsDao fruitsDao = new FruitsDaoImpl();
+        FruitRecord currentFruit = fruitsDao.getRecord(newFruit);
+        int updatedAmount = currentFruit.getAmount() - newFruit.getAmount();
+        if (updatedAmount < 0) {
+            throw new RuntimeException("It isn't possible to buy fruits!"
+                    + System.lineSeparator()
+                    + "Is available: " + currentFruit.getAmount()
+                    + " " + currentFruit.getFruit().getFruitName()
+                    + System.lineSeparator()
+                    + "Trying to buy: " + newFruit.getAmount()
+                    + " " + newFruit.getFruit().getFruitName());
+        }
+        FruitRecord fruitRecord = new FruitRecord(updatedAmount,
+                newFruit.getFruit(), newFruit.getType());
+        fruitsDao.addRecord(fruitRecord);
```

Comment:
if updated amount after subtraction is > 0, then why do we use `addRecord`?

Reply:
Fixed